lang/py/src/avro/io.py

```diff
@@ -219,6 +234,52 @@ def read_double(self):
       ((ord(self.read(1)) & 0xffL) << 56))
     return STRUCT_DOUBLE.unpack(STRUCT_LONG.pack(bits))[0]
 
+  def read_decimal_from_bytes(self, precision, scale):
+    """
+    Decimal bytes are decoded as signed short, int or long depending on the
+    size of bytes.
+    """
+    datum = self.read(self.read_long())
+    if len(datum) == 2:
+      unpacked_datum = STRUCT_SIGNED_SHORT.unpack(datum)[0]
+    elif len(datum) == 4:
+      unpacked_datum = STRUCT_SIGNED_INT.unpack(datum)[0]
+    else:
+      unpacked_datum = STRUCT_SIGNED_LONG.unpack(datum)[0]
+
+    original_prec = getcontext().prec
+    getcontext().prec = precision
+    scaled_datum = Decimal(unpacked_datum) / Decimal(pow(10, scale))
+    getcontext().prec = original_prec
+    return scaled_datum
+
+  def read_decimal_from_fixed(self, precision, scale, size):
+    """
+    Decimal is encoded as fixed. Fixed instances are encoded using the
+    number of bytes declared in the schema.
+    """
+    datum = self.read(size)
+    unscaled_datum = 0
+    msb = struct.unpack('!b', datum[0])[0]
```

Comment:
Not able to unpack as a tuple.
```
======================================================================
ERROR: test_round_trip (__main__.TestIO)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_io.py", line 220, in test_round_trip
    round_trip_datum = read_datum(writer, writers_schema)
  File "test/test_io.py", line 150, in read_datum
    return datum_reader.read(decoder)
  File "build/bdist.macosx-10.10-intel/egg/avro/io.py", line 559, in read
    return self.read_data(self.writers_schema, self.readers_schema, decoder)
  File "build/bdist.macosx-10.10-intel/egg/avro/io.py", line 606, in read_data
    writers_schema.size
  File "build/bdist.macosx-10.10-intel/egg/avro/io.py", line 263, in read_decimal_from_fixed
    msb, _ = struct.unpack('!b', datum[0])
ValueError: need more than 1 value to unpack
```
Shouldn't [this documentation](https://docs.python.org/2/library/struct.html#struct.unpack) be sufficient?